strawberry/experimental/pydantic/fields.py

```diff
@@ -25,6 +32,10 @@
         TypingGenericAlias = ()
     else:
         raise
+    if sys.version_info < (3, 10):
```

Comment:
Hi, not sure if anyone will read this comment, but it is related to this change I believe.

On Python 3.9 I'm noticing some issues here when (indirectly)  importing from this module:

```
from strawberry.experimental.pydantic.conversion_types import StrawberryTypeFromPydantic
/usr/local/lib/python3.9/site-packages/strawberry/experimental/pydantic/__init__.py:3: in <module>
    from .object_type import input, interface, type
/usr/local/lib/python3.9/site-packages/strawberry/experimental/pydantic/object_type.py:31: in <module>
    from strawberry.experimental.pydantic.fields import replace_types_recursively
/usr/local/lib/python3.9/site-packages/strawberry/experimental/pydantic/fields.py: in <module>
    from types import UnionType as TypingUnionType
E   ImportError: cannot import name 'UnionType' from 'types' (/usr/local/lib/python3.9/types.py)
```

On Python 3.10 and 3.11 it works fine.

From reading the import handling here I think it makes sense because this is what happens;

Python 3.8 -> ImportError on UnionType
* TypingGenericAlias = ()
* TypingUnionType = ()

Python 3.9 -> ImportError on UnionType
* Re-raise error

Python 3.10 -> (no import error)
* TypingGenericAlias = typing.GenericAlias
* TypingUnionType = types.UnionType